pkg/skaffold/debug/transform.go

```diff
@@ -200,3 +213,24 @@ func encodeConfigurations(configurations map[string]map[string]interface{}) stri
 	}
 	return string(bytes)
 }
+
+func describe(obj runtime.Object) (group, version, kind, description string) {
+	// get metadata/name; shamelessly stolen from from k8s.io/cli-runtime/pkg/printers/name.go
+	name := "<unknown>"
+	if acc, err := meta.Accessor(obj); err == nil {
```

Comment:
We can add test here by creating mock objects like:
```
type FakeObject struct {
     kind schema.ObjectKind
}

func (f *FakeObject) GetObjectKind() schema.ObjectKind {
   return f.kind
}

func (f *FakeObject) DeepCopyObject() {
  return f
}

type FakeObjectKind struct {
     group string
     kind string
     version string
}

func (o *FakeObjectKind) SetGroupVersionKind(kind GroupVersionKind) {
    fmt.Errorf("nyi")
}
func (o *FakeOjectKind) GroupVersionKind() schema.GroupVersionKind {
  return schema.GroupVersionKind {
     Group: o.group, 
     Version: o.version, 
     Kind: o.kind,
}
}

```

We can pass this object to `describe` and see if we get expected results.

Thanks
Tejal

Reply:
I did write some tests, but the setup was getting hairy.

/me hangs head in shame 😭

Reply:
I pushed an alternative version that uses real resources.